pkg/reconciler/pipelinerun/cancel_test.go

```diff
@@ -104,46 +110,189 @@ func TestCancelPipelineRun(t *testing.T) {
+		ignoredRuns: []*v1alpha1.Run{
+			{ObjectMeta: metav1.ObjectMeta{Name: "r3"}},
+			{ObjectMeta: metav1.ObjectMeta{Name: "r4"}},
+		},
+	}, {
+		name: "unknown-kind-on-child-references",
+		pipelineRun: &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pipeline-run-cancelled"},
+			Spec: v1beta1.PipelineRunSpec{
+				Status: v1beta1.PipelineRunSpecStatusCancelled,
+			},
+			Status: v1beta1.PipelineRunStatus{PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+				ChildReferences: []v1beta1.ChildStatusReference{{
+					TypeMeta:         runtime.TypeMeta{Kind: "InvalidKind"},
+					Name:             "t1",
+					PipelineTaskName: "task-1",
+				}},
+			}},
+		},
+		expectedErr: errors.New("error(s) from cancelling TaskRun(s) from PipelineRun test-pipeline-run-cancelled: unknown or unsupported kind `InvalidKind` for cancellation"),
 	}}
 	for _, tc := range testCases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
+			var seedTaskRuns []*v1beta1.TaskRun
+			var seedRuns []*v1alpha1.Run
+
+			for _, tr := range tc.taskRuns {
+				seedTaskRuns = append(seedTaskRuns, &v1beta1.TaskRun{
+					ObjectMeta: metav1.ObjectMeta{Name: tr.Name,
+						Labels: map[string]string{"shouldCancel": "true"}},
+				})
+			}
+			for _, tr := range tc.ignoredTaskRuns {
+				seedTaskRuns = append(seedTaskRuns, &v1beta1.TaskRun{
+					ObjectMeta: metav1.ObjectMeta{Name: tr.Name,
+						Labels: map[string]string{"shouldCancel": "false"}},
+				})
+			}
+
+			for _, r := range tc.runs {
+				seedRuns = append(seedRuns, &v1alpha1.Run{
+					ObjectMeta: metav1.ObjectMeta{Name: r.Name,
+						Labels: map[string]string{"shouldCancel": "true"}},
+				})
+			}
+			for _, r := range tc.ignoredRuns {
+				seedRuns = append(seedRuns, &v1alpha1.Run{
+					ObjectMeta: metav1.ObjectMeta{Name: r.Name,
+						Labels: map[string]string{"shouldCancel": "false"}},
+				})
+			}
+
 			d := test.Data{
 				PipelineRuns: []*v1beta1.PipelineRun{tc.pipelineRun},
-				TaskRuns:     tc.taskRuns,
-				Runs:         tc.runs,
+				TaskRuns:     seedTaskRuns,
+				Runs:         seedRuns,
 			}
 			ctx, _ := ttesting.SetupFakeContext(t)
 			ctx, cancel := context.WithCancel(ctx)
 			defer cancel()
 			c, _ := test.SeedTestData(t, ctx, d)
-			if err := cancelPipelineRun(ctx, logtesting.TestLogger(t), tc.pipelineRun, c.Pipeline); err != nil {
-				t.Fatal(err)
-			}
-			// This PipelineRun should still be complete and false, and the status should reflect that
-			cond := tc.pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
-			if cond.IsTrue() {
-				t.Errorf("Expected PipelineRun status to be complete and false, but was %v", cond)
-			}
-			if tc.taskRuns != nil {
-				l, err := c.Pipeline.TektonV1beta1().TaskRuns("").List(ctx, metav1.ListOptions{})
+
+			err := cancelPipelineRun(ctx, logtesting.TestLogger(t), tc.pipelineRun, c.Pipeline)
+			if tc.expectedErr != nil {
+				if err == nil {
+					t.Fatalf("expected error '%s', but did not get an error", tc.expectedErr)
```

Comment:
I'd use t.Errorf here and below

Reply:
Yup, those are reasonable changes, doing so.

Reply:
Actually, check that, this one does make sense as `Fatalf`, since the next check includes `err.Error()`, which would nil pointer on us if `err` is nil. I'll change the second one, but I'm leaving this one.

Reply:
I'd swap to a bool wantErr (see comment above), which allows you to simplify:
```
if tc.wantErr != (err != nil) {
  t.Errorf("expected err: %s, but got err %s")
}
```